test2code/src/main/kotlin/com/epam/drill/plugins/test2code/coverage/CoverageManager.kt

```diff
@@ -0,0 +1,78 @@
+package com.epam.drill.plugins.test2code.coverage
+
+import com.epam.drill.jacoco.AgentProbes
+import com.epam.drill.plugins.test2code.common.api.DEFAULT_TEST_NAME
+
+/**
+ * Simple probe provider that employs a lock-free map for runtime data storage.
+ * This class is intended to be an ancestor for a concrete probe array provider object.
+ * The provider must be a Kotlin singleton object, otherwise the instrumented probe calls will fail.
+ */
+open class CoverageManager(
+    // TODO EPMDJ-8256 When application is async we must use this implementation «com.alibaba.ttl.TransmittableThreadLocal»
+    private val requestThreadLocal: ThreadLocal<ExecData?> = ThreadLocal(),
+    private val globalExecData: ExecData = ExecData(),
+    private val execDataPool: DataPool<SessionTestKey, ExecData> = ConcurrentDataPool(),
+
+    private val probesProvider: ProbesProvider = SimpleProbesProvider(requestThreadLocal, globalExecData),
+    private val probesDescriptorProvider: ProbesDescriptorProvider = ConcurrentProbesDescriptorProvider(),
+    private val coverageRecorder: CoverageRecorder = ThreadCoverageRecorder(
+        execDataPool,
+        requestThreadLocal,
+        probesDescriptorProvider
+    ),
+    private val coverageSender: CoverageSender = IntervalCoverageSender(coverageRecorder, 2000L)
```

Comment:
add //TODO to replace magic number with configurable parameter

Reply:
add //TODO - do not pass coverageRecorder, but pass lambda to obtain data-to-be-sent instead